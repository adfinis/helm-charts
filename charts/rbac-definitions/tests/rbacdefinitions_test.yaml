suite: RBACDefinitions tests
templates:
  - rbacdefinitions.yaml
tests:
  - it: should match snapshot
    values:
      - ../ci/default-values.yaml
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}
  - it: should only create one RBACDefinition
    values:
      - ../examples/enabled-and-disabled.yaml
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}
  - it: should create one RBACDefinition
    values:
      - ../examples/clusterrolebinding.yaml
    asserts:
      - hasDocuments:
          count: 2
      - matchSnapshot: {}
  - it: should create one RBACDefinition with a namespace selector
    values:
      - ../examples/namespaceselector.yaml
    asserts:
      - hasDocuments:
          count: 1
      - documentIndex: 0
        equal:
          path: metadata.name
          value: devs-edit
      - documentIndex: 0
        equal:
          path: rbacBindings[0].name
          value: devs
      - documentIndex: 0
        equal:
          path: rbacBindings[0].subjects[0].name
          value: devs
      - documentIndex: 0
        equal:
          path: rbacBindings[0].roleBindings[0].clusterRole
          value: edit
      - matchSnapshot: {}
