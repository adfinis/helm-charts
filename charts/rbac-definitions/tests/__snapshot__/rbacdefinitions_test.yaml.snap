should create one RBACDefinition:
  1: |
    apiVersion: rbacmanager.reactiveops.io/v1beta1
    kind: RBACDefinition
    metadata:
      labels:
        app.kubernetes.io/app: rbac-definitions
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1beta1
        helm.sh/chart: rbac-definitions-0.1.0
      name: cluster-editors
    rbacBindings:
    - clusterRoleBindings:
      - clusterRole: edit
      name: editors
      subjects:
      - kind: User
        name: maxime@example.com
  2: |
    apiVersion: rbacmanager.reactiveops.io/v1beta1
    kind: RBACDefinition
    metadata:
      labels:
        app.kubernetes.io/app: rbac-definitions
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1beta1
        helm.sh/chart: rbac-definitions-0.1.0
      name: cluster-admins
    rbacBindings:
    - clusterRoleBindings:
      - clusterRole: cluster-admin
      name: admins
      subjects:
      - kind: User
        name: jane@example.com
should create one RBACDefinition with a namespace selector:
  1: |
    apiVersion: rbacmanager.reactiveops.io/v1beta1
    kind: RBACDefinition
    metadata:
      labels:
        app.kubernetes.io/app: rbac-definitions
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1beta1
        helm.sh/chart: rbac-definitions-0.1.0
      name: devs-edit
    rbacBindings:
    - name: devs
      roleBindings:
      - clusterRole: edit
        namespaceSelector:
          matchLabels:
            k8s.example.com/owner: devs
      subjects:
      - kind: Group
        name: devs
should match snapshot:
  1: |
    apiVersion: rbacmanager.reactiveops.io/v1beta1
    kind: RBACDefinition
    metadata:
      labels:
        app.kubernetes.io/app: rbac-definitions
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1beta1
        helm.sh/chart: rbac-definitions-0.1.0
      name: ci-example
    rbacBindings:
    - clusterRoleBindings:
      - clusterRole: edit
      name: edit
      subjects:
      - kind: User
        name: ci@example.com
should only create one RBACDefinition:
  1: |
    apiVersion: rbacmanager.reactiveops.io/v1beta1
    kind: RBACDefinition
    metadata:
      labels:
        app.kubernetes.io/app: rbac-definitions
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1beta1
        helm.sh/chart: rbac-definitions-0.1.0
      name: cluster-admins
    rbacBindings:
    - clusterRoleBindings:
      - clusterRole: cluster-admin
      name: admins
      subjects:
      - kind: User
        name: jane@example.com
